shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -220,57 +197,513 @@ - (CGRect)accessibilityFrame {
+  return [[self textInputSurrogate] comparePosition:position toPosition:other];
+}
+
+- (NSInteger)offsetFromPosition:(UITextPosition*)from toPosition:(UITextPosition*)toPosition {
+  return [[self textInputSurrogate] offsetFromPosition:from toPosition:toPosition];
+}
+
+- (UITextPosition*)positionWithinRange:(UITextRange*)range
+                   farthestInDirection:(UITextLayoutDirection)direction {
+  return [[self textInputSurrogate] positionWithinRange:range farthestInDirection:direction];
+}
+
+- (UITextRange*)characterRangeByExtendingPosition:(UITextPosition*)position
+                                      inDirection:(UITextLayoutDirection)direction {
+  return
+      [[self textInputSurrogate] characterRangeByExtendingPosition:position inDirection:direction];
+}
+
+- (UITextWritingDirection)baseWritingDirectionForPosition:(UITextPosition*)position
+                                              inDirection:(UITextStorageDirection)direction {
+  return [[self textInputSurrogate] baseWritingDirectionForPosition:position inDirection:direction];
+}
+
+- (void)setBaseWritingDirection:(UITextWritingDirection)writingDirection
+                       forRange:(UITextRange*)range {
+  [[self textInputSurrogate] setBaseWritingDirection:writingDirection forRange:range];
+}
+
+- (CGRect)firstRectForRange:(UITextRange*)range {
+  return [[self textInputSurrogate] firstRectForRange:range];
+}
+
+- (CGRect)caretRectForPosition:(UITextPosition*)position {
+  return [[self textInputSurrogate] caretRectForPosition:position];
+}
+
+- (UITextPosition*)closestPositionToPoint:(CGPoint)point {
+  return [[self textInputSurrogate] closestPositionToPoint:point];
+}
+
+- (UITextPosition*)closestPositionToPoint:(CGPoint)point withinRange:(UITextRange*)range {
+  return [[self textInputSurrogate] closestPositionToPoint:point withinRange:range];
+}
+
+- (NSArray*)selectionRectsForRange:(UITextRange*)range {
+  return [[self textInputSurrogate] selectionRectsForRange:range];
+}
+
+- (UITextRange*)characterRangeAtPoint:(CGPoint)point {
+  return [[self textInputSurrogate] characterRangeAtPoint:point];
+}
+
+- (void)insertText:(NSString*)text {
+  [[self textInputSurrogate] insertText:text];
+}
+
+- (void)deleteBackward {
+  [[self textInputSurrogate] deleteBackward];
+}
+
+#pragma mark - UIKeyInput overrides
+
+- (BOOL)hasText {
+  return [[self textInputSurrogate] hasText];
+}
+
+@end
+
+@implementation FlutterSemanticsObject {
+}
+
+#pragma mark - Override base class designated initializers
+
+// Method declared as unavailable in the interface
+- (instancetype)init {
+  [self release];
+  [super doesNotRecognizeSelector:_cmd];
+  return nil;
+}
+
+#pragma mark - Designated initializers
+
+- (instancetype)initWithBridge:(shell::AccessibilityBridge*)bridge uid:(int32_t)uid {
+  self = [super initWithBridge:bridge uid:uid];
+  return self;
+}
+
+#pragma mark - UIAccessibility overrides
+
+- (UIAccessibilityTraits)accessibilityTraits {
```

Comment:
With this implementation, we currently don't support text fields that at the same time want to be "checked" or "selected" or express any other accessibility trait other than being a text field.

If that has to be that way, we should at least document that the SemanticsFlag.isTextField on iOS is mutually exclusive with many other flags.

Or, could we just or in the other accessibility traits in the TextInputSemanticsObject?